python/cuproj/pyproject.toml

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) 2022-2023, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+[build-system]
+build-backend = "setuptools.build_meta"
+requires = [
+    "cmake>=3.26.4",
+    "cython>=0.29,<0.30",
+    "ninja",
+    "rmm==23.8.*",
+    "scikit-build>=0.13.1",
+    "setuptools",
+    "wheel",
+] # This list was generated by `rapids-dependency-file-generator`. To make changes, edit ../../dependencies.yaml and run `rapids-dependency-file-generator`.
+
+[project]
+name = "cuproj"
+version = "23.8.0"
+description = "cuProj: GPU-Accelerated Coordinate Projection"
+readme = { file = "README.md", content-type = "text/markdown" }
+authors = [
+    { name = "NVIDIA Corporation" },
+]
+license = { text = "Apache 2.0" }
+requires-python = ">=3.9"
+dependencies = [
+    "cupy-cuda11x>=12.0.0",
+    "cuspatial==23.8.*",
```

Comment:
I might have missed something but I thought it was said earlier that cuproj did not depend on cuspatial. Is this dependency correct?

Reply:
cuspatial is used in cuproj pytests, but cuspatial is not required to build or to run it.

Reply:
Okay. We need to make a list of testing dependencies and list it there, instead. See example: 

This dependencies list:
https://github.com/rapidsai/cudf/blob/1a75b3c4301107bd874e09dcf33a24cfb288c7f9/dependencies.yaml#L85-L93

with these packages:
https://github.com/rapidsai/cudf/blob/1a75b3c4301107bd874e09dcf33a24cfb288c7f9/dependencies.yaml#L558

renders into this set of optional test dependencies in pyproject.toml:
https://github.com/rapidsai/cudf/blob/1a75b3c4301107bd874e09dcf33a24cfb288c7f9/python/cudf/pyproject.toml#L55-L72

Reply:
OK thanks, trying now.

Reply:
Done.